_overviews/core/custom-collections.md

```diff
@@ -622,43 +655,31 @@ final class RNA private (
     Base.fromInt(groups(idx / N) >> (idx % N * S) & M)
   }
 
-  // Mandatory implementation of `fromSpecificIterable` and
-  // `newSpecificBuilder`, from `IterableOps`
-  override protected def fromSpecificIterable(coll: Iterable[Base]): RNA =
+  // Mandatory overrides of `fromSpecific`, `newSpecificBuilder`,
+  // and `empty`, from `IterableOps`
+  override protected def fromSpecific(coll: IterableOnce[Base]): RNA =
     RNA.fromSpecific(coll)
   override protected def newSpecificBuilder: mutable.Builder[Base, RNA] =
     RNA.newBuilder
+  override def empty: RNA = RNA.empty
 
   // Overloading of `appended`, `prepended`, `appendedAll`, `prependedAll`,
   // `map`, `flatMap` and `concat` to return an `RNA` when possible
+  def concat(suffix: IterableOnce[Base]): RNA =
+    strictOptimizedConcat(suffix, newSpecificBuilder)
+  @inline final def ++ (suffix: IterableOnce[Base]): RNA = concat(suffix)
   def appended(base: Base): RNA =
     (newSpecificBuilder ++= this += base).result()
   def appendedAll(suffix: Iterable[Base]): RNA =
-    (newSpecificBuilder ++= this ++= suffix).result()
+    strictOptimizedConcat(suffix, newSpecificBuilder)
   def prepended(base: Base): RNA =
     (newSpecificBuilder += base ++= this).result()
   def prependedAll(prefix: Iterable[Base]): RNA =
     (newSpecificBuilder ++= prefix ++= this).result()
-  def map(f: Base => Base): RNA = {
-    var b = newSpecificBuilder
-    for (base <- this) {
-      b += f(base)
-    }
-    b.result()
-  }
-  def flatMap(f: Base => IterableOnce[Base]): RNA = {
-    var b = newSpecificBuilder
-    for (base <- this) {
-      b ++= f(base)
-    }
-    b.result()
-  }
-  def concat(suffix: IterableOnce[Base]): RNA = {
-    val b = newSpecificBuilder
-    b ++= this
-    b ++= suffix
-    b.result()
-  }
+  def map(f: Base => Base): RNA =
+    strictOptimizedMap(newSpecificBuilder, f)
+  def flatMap(f: Base => IterableOnce[Base]): RNA =
+    strictOptimizedFlatMap(newSpecificBuilder, f)
```

Comment:
It would be good to consistently provide `strictOptimizedXxx` operations for all overridden operations. We cover them only partially (ie, we have `strictOptimizedMap` but not `strictOptimizedAppended`).

Reply:
I’ve created https://github.com/scala/bug/issues/11477 for that.